cmd/rook/ceph/mon.go

```diff
@@ -72,7 +72,8 @@ func startMon(cmd *cobra.Command, args []string) error {
 
 	// at first start the local monitor needs to be added to the list of mons
 	clusterInfo.Monitors = mon.ParseMonEndpoints(cfg.monEndpoints)
-	clusterInfo.Monitors[monName] = mon.ToCephMon(monName, cfg.networkInfo.PublicAddr, monPort)
+	netInfo := cfg.networkInfo.Simplify()
```

Comment:
the `Simplify()` function is called during `createContext()` further down.  I'm wondering if there is a better or more general way to not miss things like this in the future.  Do you think there should be a general reusable function like `handleRenamedFlags()`?  In that function, it would call `cfg.NetworkInfo.Simplify()` and any other renamed flags in the future could have their values loaded/read there as well all in one place.

This function could be added to every ceph command, similar to how every function gets `SetLogLevel()` and `LogStartupInfo()`.  That way its less likely to have to remember to add this Simplify call as needed in the future, it feels like we'd hit this again if we don't deal with it in a more general way.  e.g., the osd cmd already has another issue with this I can see in the rootfs/osd-per-pod-all brach :(

Reply:
@travisn let's please remember to handle this logic in the rootfs/osd-per-pod-all branch too after this is merged.